packages/@aws-cdk/aws-secretsmanager/lib/secret.ts

```diff
@@ -331,9 +331,9 @@ abstract class SecretBase extends Resource implements ISecret {
 
     if (this.encryptionKey) {
       // @see https://docs.aws.amazon.com/kms/latest/developerguide/services-secrets-manager.html
-      this.encryptionKey.grantDecrypt(
-        new kms.ViaServicePrincipal(`secretsmanager.${Stack.of(this).region}.amazonaws.com`, grantee.grantPrincipal),
-      );
+      const principal =
```

Comment:
I'm not sure this is what we need to do here. I _think_ the issue is that we are not giving enough permissions to the `grantPrincipal` on the key. Based on the [docs](https://docs.aws.amazon.com/secretsmanager/latest/userguide/security-encryption.html#security-encryption-authz) it looks like we need to add both `Decrypt` _and_ `GenerateDataKey` if we are granting permission via a service principal.

This also might be a good use case to use the new integration test assertions to _prove_ that the access works.
https://docs.aws.amazon.com/cdk/api/v2/docs/integ-tests-alpha-readme.html#assertions

Reply:
@corymhall Let me take a look